### ISSUE TITLE
Build of dependencies  getting skipped between  Debug and Release builds

### ISSUE THREAD

**Issue by FrogsInSpace** [IfcOpenShell/IfcOpenShell#5831]
Looks like commit f91fedf69ca4d744239f1c9036eecf7f3999c4b0 , which introduced checks for existing dependency installs, doesn't discriminate between existing debug and release versions of the dependencies
This causes release builds of dependencies getting skipped if debug versions exist (are installed ) and vice versa.
I guess this only affects builds triggered using the batch ( *.cmd, *.bat ) files

A solution would be setting %DEBUG_OR_RELEASE_LOWERCASE% env variable accordingly prior calling vs-cfg.cmd and inserting -%DEBUG_OR_RELEASE_LOWERCASE% to line 249 , where the INSTALL_DIR variable set. The build then would generate seperated dependency install folders

I currently run this changes through my pipeline in both modes ( Debug, Release) and will report wether this indeed fixes the issue

**Comment by FrogsInSpace** [comment]
Na ... seems installing dependencies to seperate INSTALL_DIR's leads to other build scripts failing/picking wrong INSTALL_DIR . It's propably better to improve the directory checks in build-deps.cmd to respect current Release/Debug option

**Comment by aothms** [comment]
> Na ... seems installing dependencies to seperate INSTALL_DIR's leads to other build scripts failing/picking wrong INSTALL_DIR 

Yeah all this might need some rework. As a long term goal it would also be nice if the win and nix build scripts converge a bit in approach. Or maybe the nix build script in python is a better basis for the win build script anyway, because the win .bat language is not without its quirks. Feel free to suggest sth or just incrementally fix things as you see fit.

**Comment by RickBrice** [comment]
Is this related to #4584?

**Comment by FrogsInSpace** [comment]
> Is this related to #4584?

not directly with commit f91fedf69ca4d744239f1c9036eecf7f3999c4b0 i think, but the core problem is the same: debug and release versions of some built libraries are not seperated in the install folder. I started to improve some of that in the build_deps.cmd script, lets see how far i will get ...
